docs/content/examples/map-editor.tsx

```diff
@@ -14,17 +23,82 @@ const viewer = {
   username: 'demo-maxima',
 } as const;
 
+const toMapObject = (data: string | null): MapObject | null => {
+  if (!data) {
+    return null;
+  }
+
+  try {
+    return JSON.parse(data);
+  } catch {
+    return null;
+  }
+};
```

Comment:
I feel like we should actually parse and validate each field here to match the shape of `MapObject`, otherwise if data input is wrong it will fail later in the code and is confusing.

Reply:
Do you have a preferred strategy for the validation? We could introduce something like `zod` or just add some lightweight validation for each field checking if it's present and if not, returning a default/empty value.